### PR TITLE
FEAT: add hashbytes function to string and binary expressions

### LINKED ISSUE #2310: ENH: Hash function for BigQuery
I'd like to implement the [hash function](https://ibis-project.org/docs/generated/ibis.expr.api.ValueExpr.hash.html#ibis.expr.api.ValueExpr.hash) for BigQuery (specifically, I'd like to use the [BigQuery SHA256 hash function](https://cloud.google.com/bigquery/docs/reference/standard-sql/hash_functions#sha256)), but I see a couple of questions that need to be answered before implementing.

- Currently, hash is a ValueExpr, but in BigQuery only BYTES / STRING are acceptable as inputs. Should we implement it in BigQuery as a string expression instead?
- Currently, hash outputs int64, but BigQuery outputs BYTES (which makes sense since SHA256 generates a 256-bit hash). Would it make sense to change the impala return type to match? Or is it okay to be inconsistent?

Note: other databases tend to align with how BigQuery handles hashes. See:

- [SQL Server / T-SQL](https://docs.microsoft.com/en-us/sql/t-sql/functions/hashbytes-transact-sql?view=sql-server-ver15)
- [PostgreSQL](https://www.postgresql.org/docs/11/functions-binarystring.html)

### PR DESCRIPTION
**Pull request by tswast**
Also, support `farm_fingerprint` hash type in the BigQuery engine.

Closes #2310

### CODE REVIEW

File: ibis/backends/base_sql/__init__.py [hunk 1]
```diff
@@ -371,6 +371,24 @@ def hash(translator, expr):
...
 
 
+def hashbytes(translator, expr):
+    op = expr.op()
+    arg, how = op.args
+
+    arg_formatted = translator.translate(arg)
+
+    if how == 'md5':
+        return 'md5({})'.format(arg_formatted)
```
Comment (jreback) [inline ibis/backends/base_sql/__init__.py:381]: these can be f-strings, but nbd.

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm. (minor comment). are these available on other sql backends? if so would love to have a followup which implements. pandas implements here: https://pandas.pydata.org/pandas-docs/dev/reference/api/pandas.util.hash_pandas_object.html?highlight=hash#pandas.util.hash_pandas_object

**Review by jreback — CHANGES_REQUESTED** [review]
can you add a whatsnew note and ping on green.

**Comment by jreback** [comment]
can u merge master

**Comment by tswast** [comment]
@jreback Merged! The build is green now.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @tswast very nice

**Pull request merged**